design/plugin-backup-and-restore-progress-design.md

```diff
@@ -0,0 +1,448 @@
+# Progress reporting for backups and restores handled by volume snapshotters
+
+Users face difficulty in knowing the progress of backup/restore operations of volume snapshotters. This is very similar to the issues faced by users to know progress for restic backup/restore, like, estimation of operation, operation in-progress/hung etc.
+
+Each plugin might be providing a way to know the progress, but, it need not uniform across the plugins.
+
+Even though plugins provide the way to know the progress of backup operation, this information won't be available to user during restore time on the destination cluster.
+
+So, apart from the issues like progress, status of operation, volume snapshotters have unique problems like
+- not being uniform across plugins
+- not knowing the backup information during restore operation
+- need to be optional as few plugins may not have a way to provide the progress information
+
+This document proposes an approach for plugins to follow to provide backup/restore progress, which can be used by users to know the progress.
+
+## Goals
+
+- Provide uniform way of visibility into backup/restore operations performed by volume snapshotters
+
+## Non Goals
+- Plugin implementation for this approach
+
+## Background
+
+(Omitted, see introduction)
+
+## High-Level Design
+
+### Progress of backup operation handled by volume snapshotter
+
+Progress will be updated by volume snapshotter in VolumePluginBackup CR which is specific to that backup operation.
+
+### Progress of restore operation handled by volume snapshotter
+
+Progress will be updated by volume snapshotter in VolumePluginRestore CR which is specific to that restore operation.
+
+## Detailed Design
+
+### Approach 1
+
+Existing `Snapshot` Go struct from `volume` package have most of the details related to backup operation performed by volumesnapshotters.
+This struct also gets backed up to backup location. But, this struct doesn't get synced on other clusters at regular intervals.
+It will be synced only during restore operation, and velero CLI shows few of its contents.
+
+At a high level, in this approach, this struct will be converted to a CR by adding new fields (related to Progress tracking) to it, and gets rid of `volume.Snapshot` struct.
+
+Instead of backing up of Go struct, CRs will be backed up to backup location, and it gets synced into other cluster by backupSyncController running in that cluster.
```

Comment:
Line 47 contradicts line 41. Is the plan to sync the CR on a regular basis, or at restore time?

Reply:
First para i.e., lines 41, 42 and 43 are related to current behavior. I will update it.
At line 47 is the proposed solution. So, plan is to sync the CR on regular basis.

Reply:
updated the text